bin/echo.c

```diff
@@ -70,6 +70,53 @@ static int wait_for_event(int fd, s2n_blocked_status blocked)
     return S2N_SUCCESS;
 }
 
+int early_data_recv(struct s2n_connection *conn)
+{
+    uint32_t max_early_data_size = 0;
+    GUARD_RETURN(s2n_connection_get_max_early_data_size(conn, &max_early_data_size), "Error getting max early data size");
+    if (max_early_data_size == 0) {
+        return S2N_SUCCESS;
+    }
+
+    ssize_t total_data_recv = 0;
+    ssize_t data_recv = 0;
+    bool server_success = 0;
+    s2n_blocked_status blocked = 0;
+    uint8_t *early_data_received = malloc(max_early_data_size);
+
+    do {
+        server_success = (s2n_recv_early_data(conn, early_data_received + total_data_recv,
+                max_early_data_size - total_data_recv, &data_recv, &blocked) >= S2N_SUCCESS);
+        total_data_recv += data_recv;
+    } while (!server_success);
+
+    if (total_data_recv > 0) {
+        fprintf(stdout, "Early Data received:\n");
+        for (size_t i = 0; i < total_data_recv; i++) {
+            fprintf(stdout, "%c", early_data_received[i]);
+        }
+    }
```

Comment:
You probably want to print a newline at the very end here. And wouldn't a newline after "Early data received" make it harder to detect early data?
```suggestion
    if (total_data_recv > 0) {
        fprintf(stdout, "Early Data received: ");
        for (size_t i = 0; i < total_data_recv; i++) {
            fprintf(stdout, "%c", early_data_received[i]);
        }
        fprintf(stdout, "\n");
    }
```

Reply:
Newline afterwards causes an extra newline in stdout. But I can get rid of the first newline.